dipy/reconst/dki.py

```diff
@@ -19,13 +19,352 @@
+    E2 = X*Y - 6.*Z*Z
+    E3 = (3.*X*Y - 8.*Z*Z) * Z
+    E4 = 3. * (X*Y - Z*Z) * Z**2.
+    E5 = X * Y * Z**3.
+    RD = \
+        4**(-n) * An**(-3/2.) * \
+        (1 - 3/14.*E2 + 1/6.*E3 + 9/88.*(E2**2) - 3/22.*E4 - 9/52.*E2*E3 +
+         3/26.*E5) + 3*sum_term
+
+    return RD
+
+
+def _F1m(a, b, c):
+    """ Helper function that computes function $F_1$ which is required to
+    compute the analytical solution of the Mean kurtosis.
+
+    Parameters
+    ----------
+    a : ndarray
+        Array containing the values of parameter $\lambda_1$ of function $F_1$
+    b : ndarray
+        Array containing the values of parameter $\lambda_2$ of function $F_1$
+    c : ndarray
+        Array containing the values of parameter $\lambda_3$ of function $F_1$
+
+    Returns
+    -------
+    F1 : ndarray
+       Value of the function $F_1$ for all elements of the arrays a, b, and c
+
+    Notes
+    --------
+    Function $F_1$ is defined as [1]_:
+
+    .. math::
+
+        F_1(\lambda_1,\lambda_2,\lambda_3)=
+        \frac{(\lambda_1+\lambda_2+\lambda_3)^2}
+        {18(\lambda_1-\lambda_2)(\lambda_1-\lambda_3)}
+        [\frac{\sqrt{\lambda_2\lambda_3}}{\lambda_1}
+        R_F(\frac{\lambda_1}{\lambda_2},\frac{\lambda_1}{\lambda_3},1)+\\
+        \frac{3\lambda_1^2-\lambda_1\lambda_2-\lambda_2\lambda_3-
+        \lambda_1\lambda_3}
+        {3\lambda_1 \sqrt{\lambda_2 \lambda_3}}
+        R_D(\frac{\lambda_1}{\lambda_2},\frac{\lambda_1}{\lambda_3},1)-1 ]
+
+    References
+    ----------
+    .. [1] Tabesh, A., Jensen, J.H., Ardekani, B.A., Helpern, J.A., 2011.
+           Estimation of tensors and tensor-derived measures in diffusional
+           kurtosis imaging. Magn Reson Med. 65(3), 823-836
+    """
+    # Float error used to compare two floats, abs(l1 - l2) < er for l1 = l2
+    # Error is defined as three order of magnitude larger than system's epslon
+    er = np.finfo(a.ravel()[0]).eps * 1e3
+
+    # Initialize F1
+    F1 = np.zeros(a.shape)
+
+    # Only computes F1 in voxels that have all eigenvalues larger than zero
+    cond0 = _positive_evals(a, b, c)
+
+    # Apply formula for non problematic plaussible cases, i.e. a!=b and a!=c
+    cond1 = np.logical_and(cond0, np.logical_and(abs(a - b) > er,
+                                                 abs(a - c) > er))
+    if np.sum(cond1) != 0:
+        L1 = a[cond1]
+        L2 = b[cond1]
+        L3 = c[cond1]
+        RFm = carlson_rf(L1/L2, L1/L3, np.ones(len(L1)))
+        RDm = carlson_rd(L1/L2, L1/L3, np.ones(len(L1)))
+        F1[cond1] = ((L1+L2+L3) ** 2) / (18 * (L1-L2) * (L1-L3)) * \
+                    (np.sqrt(L2*L3) / L1 * RFm +
+                     (3 * L1**2 - L1*L2 - L1*L3 - L2*L3) /
+                     (3 * L1 * np.sqrt(L2*L3)) * RDm - 1)
+
+    # Resolve possible sigularity a==b
+    cond2 = np.logical_and(cond0, np.logical_and(abs(a - b) < er,
+                                                 abs(b - c) > er))
+    if np.sum(cond2) != 0:
+        L1 = a[cond2]
```

Comment:
These lines are not currently covered by tests (see here: https://travis-ci.org/nipy/dipy/jobs/75456098)